### PR TITLE
Fixes building on Alpine Linux

### PR DESCRIPTION
**Pull request by b-i-t-n**
I found there was an issue with compiling on Alpine linux. 
```
[ 72%] Building CXX object CMakeFiles/xmr-stak-cpu.dir/jpsock.cpp.o
In file included from /xmr-stak-cpu/jpsock.cpp:33:0:
/xmr-stak-cpu/socks.h: In function 'const char* sock_strerror(char*, size_t)':
/xmr-stak-cpu/socks.h:83:19: error: invalid conversion from 'int' to 'const char*' [-fpermissive]
  return strerror_r(errno, buf, len);
         ~~~~~~~~~~^~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/xmr-stak-cpu.dir/build.make:183: CMakeFiles/xmr-stak-cpu.dir/jpsock.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/xmr-stak-cpu.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

As Alpine does not use GLIBC this line is need well only the `!defined(__GLIBC__)` others are just a bonus.

### CODE REVIEW

File: socks.h [hunk 1]
```diff
@@ -76,7 +76,7 @@ inline void sock_close(SOCKET s)
 inline const char* sock_strerror(char* buf, size_t len)
 {
 	buf[0] = '\0';
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(__FreeBSD__) || !defined(_GNU_SOURCE) || !defined(__GLIBC__)
```
Comment (psychocrypt) [inline socks.h]: This `|| !defined(_GNU_SOURCE) || !defined(__GLIBC__)` look like it is a little bit to much. The change means that all compiler those are not using `GLIBC` enters this pass. Befor ethis PR only `__APPLE__` enter the pass.
I think  it should be `#if  ( defined(__APPLE__) || defined(__FreeBSD__) ) & ( !defined(_GNU_SOURCE) || !defined(__GLIBC__) )`

My suggestion should be side effect free to other OS than `APPLE` and `FreeBSD`

**Comment by psychocrypt** [comment]
Please reopen this pull request against the dev branch, we can't merge direct to the master. After you reopened this PR I can accept it.

**Comment by b-i-t-n** [comment]
Sorry I have been away. I have changed the pull request to point to your `dev` branch.

**Comment by psychocrypt** [comment]
reopened because the base branch is changes to dev

**Comment by psychocrypt** [comment]
Could you please rebase against dev and than force push again. Github is not possible to solve the merge conflicts.

**Pull request merged**